sound/soc/sof/topology.c

```diff
@@ -809,6 +809,61 @@ static int sof_parse_tokens(struct snd_soc_component *scomp,  void *object,
 				    array_size, 1, 0);
 }
 
+static void sof_w_no_wname_in_long_name(struct snd_soc_component *scomp,
```

Comment:
@jsarha why  do we need to duplicate the parsing code when we can simply do this:
```
static const struct sof_topology_token widget_kcontrol_name_token[] = {
	{SOF_TKN_COMP_NO_WNAME_IN_KCONTROL_NAME, SND_SOC_TPLG_TUPLE_TYPE_BOOL, get_token_u16,
		offsetof(struct snd_soc_dapm_widget, no_wname_in_kcontrol_name)},
};

ret = sof_parse_tokens(scomp, w, widget_kcontrol_name_token,
		       ARRAY_SIZE(widget_kcontrol_name_token), private->array,
			       le32_to_cpu(private->size));
if (ret) {
	dev_err(scomp->dev, "error: parse stream tokens failed %d\n",
		le32_to_cpu(private->size));
	return ret;
}
```

Reply:
@ranj063 , I am pretty sure that does not work with bit-fields. I would have to turn no_wname_in_kcontrol_name from 1 bit wide to one byte wide. offsetof only has one byte granularity, hasn't it?

Reply:
@jsarha any reason why we can't make it a bool or a byte instead of a 1bit value?

Reply:
Only that all the other flags in the struct are wide bit-fields, so it would a bit out of place there, and the header is generic ASoC code, not sof only.  Of course we have @lgirdwood in the team so we can ask directly, what do you say? Can I change this just unsigned char: https://github.com/thesofproject/linux/pull/4445/commits/44de36b96baad4afff31352cec62772d0bd75c8c

The other option would be adding a bit-mask to struct sof_topology_token and implementing get_token_u8_bitfield().

Reply:
@jsarha I dont think there's any restriction that this can to be a bit field in the struct. I'd modify it to a bool to keep it simple.

Reply:
Simple me... its not that hard really. I can of course do it like this:
```
static int get_w_no_wname_in_long_name(void *elem, void *object, u32 offset)
{
	struct snd_soc_tplg_vendor_value_elem *velem = elem;
	struct snd_soc_dapm_widget *w = object;

	w->no_wname_in_kcontrol_name = !!le32_to_cpu(velem->value);
	return 0;
}

static const struct sof_topology_token dapm_widget_token[] = {
	{SOF_TKN_COMP_NO_WNAME_IN_KCONTROL_NAME, SND_SOC_TPLG_TUPLE_TYPE_BOOL,
	 get_w_no_wname_in_long_name, 0 }
};
```
That would fine, would't it, @ranj063 ?

Reply:
@jsarha yes thats perfect!